pkg/hubble/relay/pool/option.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pool
+
+import (
+	"strings"
+	"time"
+
+	"github.com/cilium/cilium/pkg/defaults"
+)
+
+// defaultOptions is the reference point for default values.
+var defaultOptions = Options{
+	Target:       "unix://" + defaults.HubbleSockPath,
+	DialTimeout:  5 * time.Second,
+	RetryTimeout: 30 * time.Second,
+}
+
+// Option customizes the configuration of the Manager.
+type Option func(o *Options) error
+
+// Options stores all the configuration values for peer manager.
+type Options struct {
+	Target       string
+	DialTimeout  time.Duration
+	RetryTimeout time.Duration
+	Debug        bool
+}
```

Comment:
i don't think any of these have to be public

Reply:
They don't _have to_ be for documentation purpose it makes sense so that the user can see what the defaults are. Current version is:
```go
// DefaultOptions is the reference point for default values.
var DefaultOptions = Options{
        PeerClientBuilder: peerTypes.LocalClientBuilder{
                LocalAddress: "unix://" + defaults.HubbleSockPath,
                DialTimeout:  5 * time.Second,
        },
        ClientConnBuilder: GRPCClientConnBuilder{
                DialTimeout: 5 * time.Second,
                Options: []grpc.DialOption{
                        grpc.WithInsecure(),
                        grpc.WithBlock(),
                },
        },
        Backoff: &backoff.Exponential{
                Min:    10 * time.Second,
                Max:    12 * time.Hour,
                Factor: 2.0,
        },
        ConnCheckInterval: 2 * time.Minute,
        RetryTimeout:      30 * time.Second,
}

// Option customizes the configuration of the Manager.
type Option func(o *Options) error

// Options stores all the configuration values for peer manager.
type Options struct {
        PeerClientBuilder peerTypes.ClientBuilder
        ClientConnBuilder ClientConnBuilder
        Backoff           BackoffDuration
        ConnCheckInterval time.Duration
        RetryTimeout      time.Duration
        Debug             bool
}
```